### PR TITLE
Add(Responsive/File Drop zone): Add Responsive and File Drop zone component

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds a component that allows a children component to be responsive.

Another `Responsive` components already exists (`visx`) but, in some cases, the scrollbar is a bit broken on children component.
This also adds a file drop zone the handle files drop and files select from an explorer.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: packages/centreon-ui/src/FileDropZone/useDropzone.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import {
```
Comment (bdauria) [inline packages/centreon-ui/src/FileDropZone/useDropzone.ts:1]: I think that this hook might deserve some unit tests in order to cover what's absent from the stories

File: packages/centreon-ui/src/FileDropZone/useDropzone.test.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    act(() => {
+      result.current.handleChangeFiles({ target: { files: fileList } });
+    });
+
+    expect(mockChangeFiles).toHaveBeenCalledWith(null);
+    expect(mockResetFilesStatusAndUploadData).not.toBeCalled();
+    expect(result.current.error).toEqual(labelInvalidFileType);
+  });
+
+  it('does not changes the files and returns an error function when "handleChangeFiles" is called and the file size is too big', () => {
```
Comment (bdauria) [inline packages/centreon-ui/src/FileDropZone/useDropzone.test.ts:100]:
```suggestion
  it('does not change the files and returns an error function when "handleChangeFiles" is called and the file size is too big', () => {
```

File: packages/centreon-ui/src/FileDropZone/useDropzone.test.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        stopPropagation: jest.fn(),
+      });
+    });
+
+    expect(mockChangeFiles).toHaveBeenCalledWith(fileList);
+    expect(mockResetFilesStatusAndUploadData).toBeCalled();
+    expect(result.current.error).toEqual(null);
+  });
+
+  it('does not changes the files and returns an error function when "handleChangeFiles" is called and the file extension is not valid', () => {
```
Comment (bdauria) [inline packages/centreon-ui/src/FileDropZone/useDropzone.test.ts:84]:
```suggestion
  it('does not change the files and returns an error function when "handleChangeFiles" is called and the file extension is not valid', () => {
```

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

File: packages/centreon-ui/src/Form/Inputs/index.tsx [hunk 4]
```diff
@@ -201,8 +215,18 @@ const Inputs = ({
...
             </div>
             {hasGroupTitle &&
-              R.not(R.equals(lastGroup, groupName as string)) && <Divider />}
-          </div>
+              R.not(R.equals(lastGroup, groupName as string)) && (
+                <Divider
+                  flexItem
+                  className={classes.divider}
+                  orientation={
+                    R.equals(groupDirection, GroupDirection.Horizontal)
```
Comment (bdauria) [inline packages/centreon-ui/src/Form/Inputs/index.tsx:223]: shouldn't be the other way around?

Comment (Thebarda) [reply packages/centreon-ui/src/Form/Inputs/index.tsx:223]: No. The `groupDirection` props means how groups are displayed (vertically, horizontally). But the orientation is about how the Divider is displayed.
Here, it makes sense for the prop `groupDirection` to mean about how we want to display groups

**Review by bdauria — APPROVED** [review]

**Pull request merged**